lib/model/model.go

```diff
@@ -2432,42 +2441,33 @@ func (m *model) GlobalDirectoryTree(folder, prefix string, levels int, dirsonly
 
 		f.Name = strings.Replace(f.Name, prefix, "", 1)
 
-		var dir, base string
-		if f.IsDirectory() && !f.IsSymlink() {
-			dir = f.Name
-		} else {
-			dir = filepath.Dir(f.Name)
-			base = filepath.Base(f.Name)
-		}
-
 		if levels > -1 && strings.Count(f.Name, sep) > levels {
 			return true
 		}
 
-		last := output
-		if dir != "." {
-			for _, path := range strings.Split(dir, sep) {
-				directory, ok := last[path]
-				if !ok {
-					newdir := make(map[string]interface{})
-					last[path] = newdir
-					last = newdir
-				} else {
-					last = directory.(map[string]interface{})
-				}
-			}
+		fullName := strings.Split(f.Name, sep)
+		name := fullName[len(fullName)-1]
+		path := fullName[:len(fullName)-1]
```

Comment:
I'd rather we used filepath.Dir/Base, as those functions were made for this purpose.

Reply:
That, definitely. Whenever you need to manipulate a path manually (not through `path/filepath`), a comment explaining what happens is probably a good idea.

Reply:
I didn't think those two behave as I needed in my algorithm. Now I see they do. :)
However, I still need `Dir()` to return an empty string instead of `.`. Guess I just `if(Dir()==".") {path = "" }` ?

Please, consider the comment below about `path` and `name` being slices instead of integers.

Reply:
Not sure about them being integers, but doing the if suggested sounds ok to me.

Reply:
Found out, that no `if` needed. (used tests and debugger as well to test it)

Reply:
I for some reason think you will need it, as the old code had it. 
I suspect it only comes up when you try to get the details of a file at the root, as the filesystem abstraction uses relative path resolution internally, which leads to "what is x/z/y relative to the filesystem abstraction root x/z" which returns y, which then Dir() returns "."

Reply:
In the example with `x/z/y`, the `Base()` returns `y` (as you wrote) and `Dir()` should return `x/z`. Am I missing something?

> strings.Split() will have to be replaced with a loop
> if it ends up being super ugly we can live with this

IMHO, I think ugly code will be less readable. Why should we replace `strings.Split()` with custom code if the function does everything we want for us on one line?